### PR TITLE
[AP-3684/AP-3715] CCMS non means tested injection

### PR DESCRIPTION
**Pull request by jsugarman**
## What
Add requestor and config for non means tested application submission to CCMS

[Link to spike AP-3684](https://dsdmoj.atlassian.net/browse/AP-3684)
[Link to story AP-3715](https://dsdmoj.atlassian.net/browse/AP-3715)

Non means tested applications have different attribute key and value requirements to both passported and non-passported applications. This is a minimal implementation that was successful in injecting an application into CCMS staging environment. However, more UAT testing and/or input from BA, testers and others is ultimately needed to confirm what attributes and what values are needed.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/fixtures/files/ccms_keys/merged_in_non_means_tested_keys.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+bank_acct:
+  BALANCE:
+    generate_block?: false
+    :value: "#bank_account_balance"
+    :br100_meaning: 'Bank accounts: Current Balance'
+    :response_type: currency
+    :user_defined: true
```
Comment (jsugarman) [inline spec/fixtures/files/ccms_keys/merged_in_non_means_tested_keys.yml:8]: remove

File: spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb [hunk 2]
```diff
@@ -119,8 +119,8 @@ def hard_coded_attrs
             ["other_savings", "OTHERSAVING_INPUT_T_10WP2_3A", "text", true, "."],
             ["other_savings", "OTHERSAVING_INPUT_T_10WP2_5A", "text", true, "Client Sole"],
             ["cli_premium", "CLIPREMIUM_INPUT_T_9WP2_12A", "text", true, "."],
-            ["proceeding", "PROC_UPPER_TRIBUNAL", "text", false, "No"],
-            ["proceeding", "PROC_UPPER_TRIBUNAL", "text", false, "No"],
+            ["proceeding_means", "PROC_UPPER_TRIBUNAL", "text", false, "No"],
```
Comment (colinbruce) [inline spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb:122]: Was this something you spotted?

**Review by colinbruce — COMMENTED** [review]
Some observations but looking good :+1

File: spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb (hunk 2 above)
Comment (jsugarman) [reply spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb:122]: I amended the XmlExtractor to make it clearer (IMO) what it is checking. `proceeding` entity is actually checking the proceeding in the MeansAssesment Entity, as opposed to the proceeding in the MeritsAssesment entity (typo in assessment deliberate.

**Review by colinbruce — APPROVED** [review]

File: spec/support/xml_extractor.rb [hunk 3]
```diff
@@ -25,12 +26,12 @@ class XmlExtractor
     means_proceeding_entity: %(/Envelope/Body/CaseAddRQ/Case/CaseDetails/ApplicationDetails/MeansAssesments/AssesmentResults/AssesmentDetails/AssessmentScreens/Entity[EntityName = "PROCEEDING"]),
     money_due: %(//MeansAssesments//AssesmentDetails//Entity[EntityName = "MONEY_DUE"]//Attributes/Attribute),
     national_savings: %(//MeansAssesments//AssesmentDetails//Entity[EntityName = "CLINATIONAL"]//Attributes/Attribute),
-    opponent: %(/Envelope/Body/CaseAddRQ/Case/CaseDetails/ApplicationDetails/MeritsAssesments/AssesmentResults/AssesmentDetails/AssessmentScreens/Entity[EntityName = "OPPONENT_OTHER_PARTIES"]//Attributes/Attribute),
-    other_party: %(/Envelope/Body/CaseAddRQ/Case/CaseDetails/ApplicationDetails/MeansAssesments/AssesmentResults/AssesmentDetails/AssessmentScreens/Entity[EntityName = "OPPONENT_OTHER_PARTIES"]//Attributes/Attribute),
+    opponent_merits: %(/Envelope/Body/CaseAddRQ/Case/CaseDetails/ApplicationDetails/MeritsAssesments/AssesmentResults/AssesmentDetails/AssessmentScreens/Entity[EntityName = "OPPONENT_OTHER_PARTIES"]//Attributes/Attribute),
```
[the remark this answers is not in the document]
Comment (jsugarman) [reply spec/support/xml_extractor.rb:29]: So i found this naming in the XmlExtractor spec helper rather unhelpful/confusing. Fundamentally there are similar 2 entities called OPPONENT_OTHER_PARTIES  - one in the MeansAssesment entity and one in the MeritsAssesment entity. The extractor gives the first the key name other_party and the second the name opponent

File: app/services/ccms/case_add_requestor_factory.rb [hunk 4]
```diff
@@ -3,6 +3,8 @@ class CaseAddRequestorFactory
     def self.call(submission, options)
       if submission.legal_aid_application.passported?
         Requestors::CaseAddRequestor.new(submission, options)
+      elsif submission.legal_aid_application.non_means_tested?
+        Requestors::NonMeansTestedCaseAddRequestor.new(submission, options)
```
[the remark this answers is not in the document]
Comment (jsugarman) [reply app/services/ccms/case_add_requestor_factory.rb:7]: Thanks, will look into adding something along those lines in a subsequent ticket/PR so we can do something like below

```
submission.case_add_requestor.new(options)
```

File: spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb [hunk 5]
```diff
@@ -119,8 +119,8 @@ def hard_coded_attrs
...
+            ["proceeding_means", "PROC_UPPER_TRIBUNAL", "text", false, "No"],
```
Comment (jsugarman) [inline spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_additional_property_attributes_spec.rb:123]: duplicate from line 122?

**Pull request merged**